app/setup/index.js

```diff
@@ -0,0 +1,259 @@
+                status: 'failure',
+                errorMessage: 'Plotly username can not be empty.'
+            });
+            return;
+        }
+        if (!this.state.domain && this.state.serverType === ONPREM) {
+            this.setState({
+                status: 'failure',
+                errorMessage: 'Plotly On Premise domain can not be empty.'
+            });
+            return;
+        }
+        this.setState({errorMessage: ''});
+        this.setTimeoutAuthDone();
+        if (isWebBrowser) {
+            window.open(this.buildOauthUrl(), '_blank');
+        } else {
+            dynamicRequireElectron().shell.openExternal(this.buildOauthUrl());
+        }
+    }
+
+    updateStateWithEvent(e) {
+        this.setState({[e.target.name]: e.target.value});
+    }
+
+    render() {
+        const {errorMessage, status} = this.state;
+        const submitButton = (value) => {
+            return (
+                <button
+                    onClick={(e) => this.chooseServerType(e.target.value)}
+                    className="btn btn-large btn-primary"
+                    value={value}
+                    style={{margin: '10px'}}
+                >
+                    {SERVER_TYPES[value]}
+                </button>
+            );
+        };
+
+        const loginButton = () => {
+            return (
+                <button
+                    className="btn btn-large btn-primary"
+                    style={{display: 'block', margin: 'auto'}}
+                    onClick={() => this.authenticateUser()}
+                >{'Login'}</button>
+            );
+        };
+
+        const chooseServerType = () => {
+            return (
+                <div className="control-group">
+                <h3 className="block-center-heading">
+                    Choose Your Plotly Use Case
+                </h3>
+                    <div className="controls" style={{padding: '20px'}}>
+                        {submitButton(CLOUD)}
+                        {submitButton(ONPREM)}
+                    </div>
+                    <span>
+                        {'Learn more about our services '}
+                        <a href="https://plot.ly/products/cloud/">
+                        {'here'}</a>.
+                    </span>
+                </div>
+            );
+        };
+
+        const loginCloud = () => {
+            return (
+                <div className="control-group">
+                    <h3 className="block-center-heading">{'Login Into Your Account'}</h3>
+                    <div className="controls" style={{padding: '20px'}}>
+                        <div className="form-group">
+                            <label>Your Plotly Username</label>
+                            <input
+                                type="text"
+                                className="form-control"
+                                placeholder="johndoe"
+                                name="username"
```

Comment:
we don't really use this `event.target.name` pattern anywhere else, we just explicitly set the key as part of the event, so `onChange={e => this.setState({username: e.target.value})}`